cardano-config/src/Cardano/Config/Types.hs

```diff
@@ -641,9 +640,7 @@ instance HasKESMetricsData (ShelleyBlock c) where
   getKESMetricsData protoInfo forgeState =
       TPraosKESMetricsData currKesPeriod maxKesEvos oCertStartKesPeriod
     where
-      TPraosForgeState { tpraosHotKey } = forgeState
-
-      HotKey currKesPeriod _ = tpraosHotKey
+      HotKey { hkEvolution = currKesPeriod } = chainIndepState forgeState
```

Comment:
I think this can be simplified now that `HotKey` contains all the necessary information:
```haskell
data HotKey c = HotKey {
      hkStart     :: !Absolute.KESPeriod
   ,  hkEnd       :: !Absolute.KESPeriod
      -- ^ Currently derived from `TPraosParams`:
      -- > hkEnd = hkStart + tpraosMaxKESEvo
    , hkEvolution :: !KESEvolution
      -- ^ Invariant:
      -- > hkStart + hkEvolution in [hkStart, hkEnd]
    , hkKey       :: !(SignKeyKES (KES c))
    }
```

So:
`oCertStartKesPeriod` == `hkStart`
`maxKesEvos` == `hkEnd - hkStart`

To avoid confusion between absolute and relative periods (relative to `oCertStartKesPeriod`), we use `Evolution` in consensus to refer to relative periods. A key starts at evolution 0 (corresponding to `oCertStartKesPeriod`) and is then evolved until `oCertStartKesPeriod` + `maxKesEvos` (== `hkEnd`), after which it is no longer valid.

Now that all three arguments to `TPraosKESMetricsData` can be derived from the `HotKey`, the `ProtocolInfo` argument is no longer needed, which will make it easier to define an instance for `HasKESMetricsData` for `HardForkBlock xs`.

Reply:
I have pushed a commit that implements this simplification.

I have not yet updated the `HasKESMetricsData` instance for `HardForkBlock`.